interpreter/cling/lib/Interpreter/CIFactory.cpp

```diff
@@ -1410,53 +1460,27 @@ namespace {
       argvCompile.push_back("-");
     }
 
-    auto InvocationPtr = std::make_shared<clang::CompilerInvocation>();
-
-    // The compiler invocation is the owner of the diagnostic options.
-    // Everything else points to them.
-    DiagnosticOptions& DiagOpts = InvocationPtr->getDiagnosticOpts();
     // add prefix to diagnostic messages if second compiler instance is existing
     // e.g. in CUDA mode
     std::string ExeName = "";
     if (COpts.CUDAHost)
       ExeName = "cling";
     if (COpts.CUDADevice)
       ExeName = "cling-ptx";
-    llvm::IntrusiveRefCntPtr<DiagnosticsEngine> Diags =
-        SetupDiagnostics(DiagOpts, ExeName);
-    if (!Diags) {
-      cling::errs() << "Could not setup diagnostic engine.\n";
-      return nullptr;
-    }
 
-    llvm::Triple TheTriple(llvm::sys::getProcessTriple());
-    clang::driver::Driver Drvr(argv[0], TheTriple.getTriple(), *Diags);
-    //Drvr.setWarnMissingInput(false);
-    Drvr.setCheckInputsExist(false); // think foo.C(12)
-    llvm::ArrayRef<const char*>RF(&(argvCompile[0]), argvCompile.size());
-    std::unique_ptr<clang::driver::Compilation>
-      Compilation(Drvr.BuildCompilation(RF));
-    if (!Compilation) {
-      cling::errs() << "Couldn't create clang::driver::Compilation.\n";
-      return nullptr;
-    }
+    std::unique_ptr<clang::driver::Compilation> Compilation;
 
-    const llvm::opt::ArgStringList* CC1Args = GetCC1Arguments(Compilation.get());
-    if (!CC1Args) {
-      cling::errs() << "Could not get cc1 arguments.\n";
+    auto CIOrErr = CreateCI(argvCompile, ExeName, Compilation);
+    if (!CIOrErr) {
+      cling::errs() << "Could not create CI: " << CIOrErr.takeError() << "\n";
       return nullptr;
     }
```

Comment:
see comment above, I think the function currently returns a "successful" `nullptr` and not actually an error...